app/code/Magento/InventoryCatalogSearch/Test/Integration/Model/Indexer/FulltextTest.php

```diff
@@ -0,0 +1,309 @@
+ * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+ * @magentoDataFixture ../../../../app/code/Magento/InventorySalesApi/Test/_files/stock_website_sales_channels.php
+ * @magentoDataFixture ../../../../app/code/Magento/InventoryIndexer/Test/_files/reindex_inventory.php
+ */
+class FulltextTest extends TestCase
+{
+    /**
+     * @var \Magento\Framework\Indexer\IndexerInterface
+     */
+    private $indexer;
+
+    /**
+     * @var Fulltext
+     */
+    private $resourceFulltext;
+
+    /**
+     * @var QueryFactory
+     */
+    private $queryFactory;
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var ProductInterface
+     */
+    private $productSku1;
+
+    /**
+     * @var ProductInterface
+     */
+    private $productSku2;
+
+    /**
+     * @var ProductInterface
+     */
+    private $productSku3;
+
+    /**
+     * @var  Dimension
+     */
+    private $dimension;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var string
+     */
+    private $storeCodeBefore;
+
+    /**
+     * @var Action
+     */
+    private $productAction;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+
+        /** @var \Magento\Framework\Indexer\IndexerInterface indexer */
+        $this->indexer = $objectManager->get(Indexer::class);
+        $this->indexer->load(CatalogSearchIndexer::INDEXER_ID);
+
+        $this->resourceFulltext = $objectManager->get(Fulltext::class);
+        $this->queryFactory = $objectManager->get(QueryFactory::class);
+
+        $this->dimension = $objectManager->create(
+            Dimension::class,
+            ['name' => 'scope', 'value' => '1']
+        );
+
+        $this->storeManager = Bootstrap::getObjectManager()->get(StoreManagerInterface::class);
+        $this->storeCodeBefore = $this->storeManager->getStore()->getCode();
+
+        $this->productRepository = Bootstrap::getObjectManager()->get(ProductRepositoryInterface::class);
+        $this->productSku1 = $this->productRepository->get('SKU-1');
+        $this->productSku2 = $this->productRepository->get('SKU-2');
+        $this->productSku3 = $this->productRepository->get('SKU-3');
+        $this->productAction = Bootstrap::getObjectManager()->get(Action::class);
+    }
+
+    /**
+     * @param string $store
+     * @param array $expectedSearchSize
+     * @return void
+     *
+     * @dataProvider reindexAllDataProvider
+     */
+    public function testReindexAll(string $store, array $expectedSearchSize)
+    {
+        $this->storeManager->setCurrentStore($store);
+        $this->indexer->reindexAll();
+
+        $products = $this->search('Orange');
+        $this->assertCount($expectedSearchSize[0], $products);
+        if ($expectedSearchSize[0] > 0) {
```

Comment:
Avoid having conditions in tests
Move responsibility to data provider

Reply:
Any idea how better to do it in this case?